packages/gatsby/src/commands/build-html.ts

```diff
@@ -53,11 +60,18 @@ const runWebpack = (
       stage === `develop-html`
     ) {
       devssrWebpackCompiler = webpack(compilerConfig)
+      devssrWebpackCompiler.hooks.invalid.tap(`ssr file invalidation`, file => {
+        needToRecompileSSRBundle = true
+      })
+      devssrWebpackCompiler.hooks.watchRun.tap(`ssr watch run`, file => {
+        needToRecompileSSRBundle = true
+      })
```

Comment:
This should not be needed - `watchRun` hook is when webpack compilation pass start in watching mode (including when resuming) - I only added it before to print some console message. It also doesn't really hurt, because we will get `compiler.watch` callback that clears it, but it's just a bit confusing to set it here

Unless you had a reason to add it which I'm missing?

Reply:
nope just copy/pasted :-D I'll remove it